src/ldesify.ts

```diff
@@ -189,3 +196,90 @@ export function ldesify(
         return writer.push(st);
     });
 }
+
+export function ldesify_sds(
+    reader: Stream<string>,
+    writer: Writer<string>,
+    statePath: string | undefined,
+    sourceStream: Term | undefined,
+    targetStream: Term,
+    modifiedPathM?: Term, // This are the things that are added to the new entities, not necessarily related to the real objects
+    isVersionOfPathM?: Term,
+) {
+    const modifiedPath: Term = modifiedPathM || MODIFIED;
+    const versionPath: Term = isVersionOfPathM || IS_VERSION_OF;
+
+    let cache: { [key: string]: string } = {};
+    if (statePath) cache = getPrevState(statePath);
+
+    Cleanup(() => {
+        if (statePath) {
+            writeFileSync(statePath, JSON.stringify(cache), {
+                encoding: "utf8",
+            });
+        }
+    });
+    reader.on("end", () => {
+        if (statePath) {
+            writeFileSync(statePath, JSON.stringify(cache), {
+                encoding: "utf8",
+            });
+        }
+    });
+
+    const extractor = new Extractor(new CBDShapeExtractor(), sourceStream);
+    reader.on("data", async (x) => {
+        const quads = new Parser().parse(x);
+
+        const records = await extractor.parse_records(quads);
+
+        for (const rec of records) {
+            if (rec.data.quads.length === 0) continue;
+
+            const hash = await canonize(rec.data.quads, {
+                algorithm: "RDFC-1.0",
+            });
+            if (cache[rec.data.id.value] === hash) continue;
+
+            cache[rec.data.id.value] = hash;
+
+            const date = new Date();
+            const id = new NamedNode(rec.data.id.value + date.getTime());
+
+            const quads = [
+                df.quad(
+                    id,
+                    <Quad_Predicate>modifiedPath,
+                    df.literal(date.toISOString(), XSD.terms.dateTime),
+                ),
+
+                df.quad(
+                    id,
+                    <Quad_Predicate>versionPath,
+                    <Quad_Object>rec.data.id,
+                ),
+
+                ...rec.data.quads.map((q) =>
+                    df.quad(q.subject, q.predicate, q.object, id),
```

Comment:
That's different from how it was done before, right? Now you overwrite the graph in the quads, while previously the quads were just passed along in eg. the bucketizer iirc.

Reply:
Yess, this should be added to the readme I guess.
My use case here is that nested deep objects shouldn't be ldesified but copied for that sds record
If all nested objects should be in the default_graph, things will break.

(LDES supports encapsulation with graphs)